src/components/textbox/textbox.test.tsx

```diff
@@ -0,0 +1,607 @@
+});
+
+test.each([
+  "enter",
+  "done",
+  "go",
+  "next",
+  "previous",
+  "search",
+  "send",
+] as EnterKeyHintTypes[])(
+  "'enterKeyHint' is correctly passed to the input when prop value is %s",
+  (keyHints) => {
+    render(<Textbox value="foobar" enterKeyHint={keyHints} />);
+
+    expect(screen.getByRole("textbox")).toHaveAttribute(
+      "enterkeyhint",
+      keyHints
+    );
+  }
+);
+
+test.each(["disabled", "readOnly"])(
+  "does not call iconOnClick handler for the icon when input is %s",
+  async (propName) => {
+    const onClick = jest.fn();
+    const iconOnClick = jest.fn();
+    const user = userEvent.setup();
+
+    render(
+      <Textbox
+        value="foobar"
+        inputIcon="search"
+        onClick={onClick}
+        iconOnClick={iconOnClick}
+        disabled={propName === "disabled"}
+        readOnly={propName === "readOnly"}
+      >
+        normal children
+      </Textbox>
+    );
+    await user.click(screen.getByTestId("input-icon-toggle"));
+    expect(iconOnClick).not.toHaveBeenCalled();
+  }
+);
+
+test.each(["disabled", "readOnly"])(
+  "does not call iconOnMouseDown handler for the icon when input is %s",
+  async (propName) => {
+    const onMouseDown = jest.fn();
+    const iconOnMouseDown = jest.fn();
+    const user = userEvent.setup();
+
+    render(
+      <Textbox
+        value="foobar"
+        inputIcon="search"
+        onMouseDown={onMouseDown}
+        iconOnMouseDown={iconOnMouseDown}
+        disabled={propName === "disabled"}
+        readOnly={propName === "readOnly"}
+      >
+        normal children
+      </Textbox>
+    );
+    await user.click(screen.getByTestId("input-icon-toggle"));
+    expect(iconOnMouseDown).not.toHaveBeenCalled();
+  }
+);
+
+test.each(["disabled", "readOnly"])(
+  "does not call onClick handler when input is %s and icon is clicked",
+  async (propName) => {
+    const onClick = jest.fn();
+    const user = userEvent.setup();
+
+    render(
+      <Textbox
+        value="foobar"
+        inputIcon="search"
+        onClick={onClick}
+        disabled={propName === "disabled"}
+        readOnly={propName === "readOnly"}
+      >
+        normal children
+      </Textbox>
+    );
+    await user.click(screen.getByTestId("input-icon-toggle"));
+    expect(onClick).not.toHaveBeenCalled();
+  }
+);
+
+test.each(["disabled", "readOnly"])(
+  "does not call onMouseDown handler when input is %s and icon is clicked",
+  async (propName) => {
+    const onMouseDown = jest.fn();
+    const user = userEvent.setup();
+
+    render(
+      <Textbox
+        value="foobar"
+        inputIcon="search"
+        onMouseDown={onMouseDown}
+        disabled={propName === "disabled"}
+        readOnly={propName === "readOnly"}
+      >
+        normal children
+      </Textbox>
+    );
+    await user.click(screen.getByTestId("input-icon-toggle"));
+    expect(onMouseDown).not.toHaveBeenCalled();
+  }
+);
+
+describe("validation icon", () => {
```

Comment:
suggestion(non-blocking): this describe block isn't adding much value

We should probably either be more verbose with the description or drop the describe block